tests/Stormpath/Tests/Cache/CacheTest.php

```diff
@@ -102,6 +102,52 @@ public function testWillUpdateCacheWhenResourceUpdates()
         $application->delete();
     }
 
+    public function testItHandlesExpandedResources()
+    {
+        $cache = parent::$client->dataStore->cache;
+        // Create Account and add to Group
+        $directory = \Stormpath\Resource\Directory::instantiate(array('name' => 'Main Directory' .md5(time())));
+        self::createResource(\Stormpath\Resource\Directory::PATH, $directory);
+
+        $group = \Stormpath\Resource\Group::instantiate(array('name' => 'A New Group' . md5(time())));
+
+        $directory->createGroup($group);
+
+        $account = \Stormpath\Resource\Account::instantiate(array('givenName' => 'Account Name',
+            'surname' => 'Surname',
+            'email' => md5(time()) .'@unknown12345.kot',
+            'password' => 'superP4ss'));
+
+        $directory->createAccount($account);
+
+        $groupMembership = \Stormpath\Resource\GroupMembership::instantiate();
+        $groupMembership->account = $account;
+        $groupMembership->group = $group;
+        $groupMembership = \Stormpath\Resource\GroupMembership::create($groupMembership);
+        // Check the cache
+        $acct = \Stormpath\Resource\Account::get($account->href,array('expand'=>'groups,groupMemberships'));
+        $this->assertEquals($acct->surname, 'Surname', 'one');
+
+        // Make change to the account
+        $account->surname = "Test";
+        $account->save();
+
+        $acct2 = \Stormpath\Resource\Account::get($account->href,array('expand'=>'groups,groupMemberships'));
+
+//        var_dump($cache->get($acct->href.":groups,groupMemberships"));
+//        var_dump($cache->get($acct2->href.":groups,groupMemberships"));
+        //Confirm Changes to both account cache and expanded cache
+        $this->assertEquals($acct2->surname, 'Test', 'two');
+        $this->assertEquals($account->surname, 'Test', 'three');
+
+
+
+
+        $groupMembership->delete();
+        $directory->delete();
+        $account->delete();
```

Comment:
You only need to delete the directory and all children resources will get deleted (accounts, groups, group memberships, etc.).